### PR TITLE
Fix the problem of SSheetDialog closing rendering and provide OnClose…

### PR DESCRIPTION
**Pull request by kogel-net**
…d event

# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by doddgu** [comment]
@MayueCif PTAL

File: src/Masa.Stack.Components/Shared/IntegrationComponents/SSheetDialog.razor [hunk 1]
```diff
@@ -84,5 +88,16 @@
...
+        {
+            if (OnClosed.HasDelegate)
+            {
+                NextTick(async () =>
+                {
+                    await Task.Delay(150);
+                    await OnClosed.InvokeAsync(Ref);
+                });
+            }
+        }
```
Comment (capdiem) [inline src/Masa.Stack.Components/Shared/IntegrationComponents/SSheetDialog.razor:101]:
```suggestion
        if (OnClosed.HasDelegate)
        {
            NextTick(async () =>
            {
                await Task.Delay(150);
                await OnClosed.InvokeAsync(Ref);
            });
        }
```

File: src/Masa.Stack.Components/Shared/IntegrationComponents/SSheetDialog.razor [hunk 2]
```diff
@@ -74,6 +75,9 @@
     [Parameter]
     public bool EnableDomReload { get; set; }
 
+    [Parameter]
+    public EventCallback<ElementReference> OnClosed { get; set; }
```
Comment (capdiem) [inline src/Masa.Stack.Components/Shared/IntegrationComponents/SSheetDialog.razor:79]:
```suggestion
    public EventCallback OnClosed { get; set; }
```

**Review by capdiem — CHANGES_REQUESTED** [review]

File: src/Masa.Stack.Components/Shared/IntegrationComponents/SSheetDialog.razor [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ﻿@namespace Masa.Stack.Components
+@inherits BDomComponentBase
```
Comment (capdiem) [inline src/Masa.Stack.Components/Shared/IntegrationComponents/SSheetDialog.razor:2]:
```suggestion
@inherits NextTickComponentBase
```

**Review by capdiem — CHANGES_REQUESTED** [review]

**Review by capdiem — APPROVED** [review]
LGTM

**Pull request merged**